src/loaders/sdf.cpp

```diff
@@ -0,0 +1,211 @@
+#include <fstream>
+#include <sstream>
+#include <set>
+#include <tinyxml.h>
+
+#include "tue/config/configuration.h"
+#include "tue/config/loaders/sdf.h"
+#include "tue/config/loaders/xml.h"
+#include "tue/config/read.h"
+#include "tue/config/node_type.h"
+
+#include "tue/filesystem/path.h"
+
+namespace tue
+{
+
+namespace config
+{
+
+// ----------------------------------------------------------------------------------------------------
+
+static const std::set<std::string> SDF_ARRAY_SET {"include", "link" , "collision", "visual", "joint",
+                                                 "shape", "areas"}; //temp for development.
+static const std::set<std::string> SDF_MAP_SET {"sdf", "world", "audio", "wind", "pose", "atmosphere", "gui", "camera",
+                                          "track_visual", "plugin", "model", "gripper", "geometry", "box",
+                                          "cylinder", "heightmap", "blend", "image", "mesh", "texture", "plane",
+                                          "polyline", "sphere" };
+static const std::set<std::string> SDF_VALUE_SET {"device", "linear_velocity", "uri", "name", "static", "pose", "gravity",
+                                            "magnetic_field", "type", "temperature", "pressure", "temperature_gradient",
+                                            "fullscreen", "view_controller", "projection_type", "min_dist", "max_dist",
+                                            "use_model_frame", "xyz", "inherit_yaw", "filename", "empty", "size",
+                                            "radius", "length", "pos", "diffuse", "normal", "min_height", "fade_dist",
+                                            "sampling", "point", "height" };
+
+
+// ----------------------------------------------------------------------------------------------------
+
+tue::config::NodeType getSDFNodeType (const std::string& element_name)
+{
+    std::set<std::string>::const_iterator it = SDF_ARRAY_SET.find(element_name);
+    if (it != SDF_ARRAY_SET.end())
+        return ARRAY;
+
+    it = SDF_MAP_SET.find(element_name);
+    if (it == SDF_MAP_SET.end())
+        std::cout << "Element: '" << element_name << "' not in SDF ARRAY or MAP list. Will return MAP as type." << std::endl;
+
+    return MAP;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool loadFromSDFElement(const TiXmlElement& element, ReaderWriter& config, const NodeType node_type)
+{
+    if (element.FirstChildElement() == nullptr)
+    {
+        return loadFromXMLText(element, config);
```

Comment:
Don't you skip attributes like this?

Reply:
Yep, we discussed it. I will put it in a comment